vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SnapshotManager.java

```diff
@@ -0,0 +1,370 @@
+import edu.asu.diging.vspace.core.services.IExhibitionManager;
+import edu.asu.diging.vspace.core.services.IExternalLinkManager;
+import edu.asu.diging.vspace.core.services.IModuleLinkManager;
+import edu.asu.diging.vspace.core.services.IModuleManager;
+import edu.asu.diging.vspace.core.services.ISequenceManager;
+import edu.asu.diging.vspace.core.services.ISnapshotManager;
+import edu.asu.diging.vspace.core.services.ISpaceDisplayManager;
+import edu.asu.diging.vspace.core.services.ISpaceLinkManager;
+import edu.asu.diging.vspace.core.services.ISpaceManager;
+
+@Service
+public class SnapshotManager implements ISnapshotManager {
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+    
+    @Autowired
+    private SpaceRepository spaceRepository;
+    
+    @Autowired
+    private IModuleManager moduleManager;
+
+    @Autowired
+    private SlideManager slideManager;
+
+    @Autowired
+    private ISequenceManager sequenceManager;
+    
+    @Autowired
+    @Qualifier("storageEngineDownloads")
+    private IStorageEngine storageEngineDownloads;
+    
+    @Autowired
+    private SpringTemplateEngine springTemplateEngine;
+
+    @Autowired
+    private ISpaceDisplayManager spaceDisplayManager;
+
+    @Autowired
+    private IExhibitionManager exhibitManager;
+
+    @Autowired
+    private IModuleLinkManager moduleLinkManager;
+
+    @Autowired
+    private ISpaceLinkManager spaceLinkManager;
+
+    @Autowired
+    private IExternalLinkManager externalLinkManager;
+    
+    @Autowired
+    private ISpaceManager spaceManager;
+    
+    @Autowired
+    private SnapshotTaskRepository snapshotTaskRepository;
+    
+    @Autowired
+    private IStorageManager storageManager;
+    
+    
+    public final static String IMAGES_FOLDER_NAME = "images";
+
+    private static final String RESOURCES_FOLDER_NAME = "resources";
+    
+    @Async
+    @Override
+    @Transactional
+    public void createSnapShot(String resourcesPath, String exhibitionFolderName,SequenceHistory sequenceHistory, ExhibitionDownload exhibitionDownload)  throws IOException, InterruptedException {
+        storageEngineDownloads.copyToFolder(exhibitionFolderName + File.separator + RESOURCES_FOLDER_NAME, resourcesPath);
+        List<Space> spaces= spaceRepository.findAllBySpaceStatus(SpaceStatus.PUBLISHED);
+
+        for(Space space : spaces) {
+            downloadSpace(space, exhibitionFolderName, sequenceHistory);                
+        }         
+
+        SnapshotTask snapshotTask = exhibitionDownload.getSnapshotTask();
+        snapshotTask.setTaskComplete(true);
+        snapshotTaskRepository.save(snapshotTask);   
+    }
+    
+    /**
+     * 
+     * Download given space and related modules into exhibitionFolderPath
+     * @param space
+     * @param exhibitionFolderPath
+     * @param context
+     */
+    @Override
+    public void downloadSpace(Space space, String exhibitionFolderName,  SequenceHistory sequenceHistory) {
```

Comment:
is this ever called from outside the class? If not, should be private.

Reply:
same for all other methods that are only used in this class and aren't likely to be used by themselves outside of the class